examples/osp-flatcar-aws.yaml

```diff
@@ -0,0 +1,511 @@
+            fs.inotify.max_user_watches = 1048576
+
+    - path: /opt/bin/setup_net_env.sh
+      permissions: 0755
+      content:
+        inline:
+          encoding: none
+          data: |
+            #!/usr/bin/env bash
+            echodate() {
+              echo "[$(date -Is)]" "$@"
+            }
+
+            # get the default interface IP address
+            DEFAULT_IFC_IP=$(ip -o  route get 1 | grep -oP "src \K\S+")
+
+            if [ -z "${DEFAULT_IFC_IP}" ]
+            then
+              echodate "Failed to get IP address for the default route interface"
+              exit 1
+            fi
+
+            # get the full hostname
+            FULL_HOSTNAME=$(hostname -f)
+
+            # write the nodeip_env file
+            # we need the line below because flatcar has the same string "coreos" in that file
+            if grep -q coreos /etc/os-release
+            then
+              echo -e "KUBELET_NODE_IP=${DEFAULT_IFC_IP}\nKUBELET_HOSTNAME=${FULL_HOSTNAME}" > /etc/kubernetes/nodeip.conf
+            elif [ ! -d /etc/systemd/system/kubelet.service.d ]
+            then
+              echodate "Can't find kubelet service extras directory"
+              exit 1
+            else
+              echo -e "[Service]\nEnvironment=\"KUBELET_NODE_IP=${DEFAULT_IFC_IP}\"\nEnvironment=\"KUBELET_HOSTNAME=${FULL_HOSTNAME}\"" > /etc/systemd/system/kubelet.service.d/nodeip.conf
+            fi
+
+    - path: /opt/bin/setup
+      permissions: 0755
+      content:
+        inline:
+          encoding: none
+          data: |
+            #!/bin/bash
+            set -xeuo pipefail
+            if systemctl is-active ufw; then systemctl stop ufw; fi
+            systemctl mask ufw
+
+            {{- if .FlatcarConfig.DisableUpdateEngine }}
+            cat << EOF | tee /etc/polkit-1/rules.d/60-noreboot_norestart.rules
+            polkit.addRule(function(action, subject) {
+              if (action.id == "org.freedesktop.login1.reboot" ||
+                  action.id == "org.freedesktop.login1.reboot-multiple-sessions") {
+                  if (subject.user == "core") {
+                      return polkit.Result.YES;
+                  } else {
+                      return polkit.Result.AUTH_ADMIN;
+                  }
+              }
+            });
+            EOF
+
+            systemctl stop update-engine.service
+            systemctl mask update-engine.service
+            {{- end }}
+
+            {{- if .FlatcarConfig.DisableLocksmithD }}
+            systemctl stop locksmithd.service
+            systemctl mask locksmithd.service
+            {{- end }}
+
+            {{- /* As we added some modules and don't want to reboot, restart the service */}}
+            systemctl restart systemd-modules-load.service
+            sysctl --system
+
+            {{- /* Make sure we always disable swap - Otherwise the kubelet won't start'. */}}
+            swapoff -a
+
+            {{- template "safeDownloadBinariesScript" }}
```

Comment:
Please have a dedicate service file and unit for this script not part of the setup script.